src/Hl7.Fhir.Serialization/FhirXmlNode.cs

```diff
@@ -437,5 +437,38 @@ object checkRepresentation(ITypedElement node, IExceptionSource ies, object _)
 
         private ExceptionNotification buildException(string message) => ExceptionNotification.Error(
                 new StructuralTypeException("Parser: " + message));
+
+        public string OriginalElement
+        {
+            get {
+                if (((XElement)Current.Parent).Name.Namespace != ((XElement)Current).Name.Namespace)
```

Comment:
create a local variable like this `var current = Current as XElement` and use this in this getter (instead of casting it on 3 points)

Reply:
Done.